### DISCUSSION TITLE
TransferData vs MapQuery vs ProjectUMAP

### DISCUSSION THREAD

**Discussion by royfrancis** [satijalab/seurat#5320] [Q&A] [answered]
What's the difference between `TransferData()`, `MapQuery()` and `ProjectUMAP()`? They all sounds similar to me. Can someone please explain?

**Comment by andrewwbutler** [comment, marked as answer]
Hi,

They are all functions used in "reference-based" analyses. Here is how they relate. `TransferData` is used to transfer categorical or continuous data from a reference to a query dataset - for example to classify query cell types or impute missing features. In theory, you could use the same approach to transfer UMAP coordinates but the UMAP developers have an existing strategy for placing new points into an existing UMAP embedding that is more appropriate (see [here](https://umap-learn.readthedocs.io/en/latest/transform.html)). `ProjectUMAP` is a basically a wrapper to enable that functionality. Often when running these reference-based analyses, you want to do both (e.g. predict cell type labels and generate a unified UMAP plot) and so `MapQuery` is a higher level wrapper around these both.

**Reply by royfrancis** [reply]
Thanks for the explanation. A follow-up question. So, as I understand, `ProjectUMAP()` is purely for UMAP. With TransferData() and MapQuery(), is it possible to use any projection (PCA, ICA etc..)? Or are there any limitations?
